ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/T8nSubCommand.java

```diff
@@ -0,0 +1,557 @@
+        if (coinbase != null && coinbase.isEmpty()) {
+          worldStateUpdater.deleteAccount(coinbase.getAddress());
+        }
+        final Account txSender = worldStateUpdater.getAccount(transaction.getSender());
+        if (txSender != null && txSender.isEmpty()) {
+          worldStateUpdater.deleteAccount(txSender.getAddress());
+        }
+      }
+      if (result.isInvalid()) {
+        invalidTransactions.add(
+            new RejectedTransaction(i, result.getValidationResult().getErrorMessage()));
+      } else {
+        validTransactions.add(transaction);
+
+        long transactionGasUsed = transaction.getGasLimit() - result.getGasRemaining();
+
+        gasUsed += transactionGasUsed;
+        long intrinsicGas =
+            protocolSpec
+                .getGasCalculator()
+                .transactionIntrinsicGasCost(
+                    transaction.getPayload(), transaction.getTo().isEmpty());
+        tracer.traceEndTransaction(
+            result.getOutput(), gasUsed - intrinsicGas, timer.elapsed(TimeUnit.NANOSECONDS));
+        TransactionReceipt receipt =
+            protocolSpec
+                .getTransactionReceiptFactory()
+                .create(transaction.getType(), result, worldState, gasUsed);
+        Bytes gasUsedInTransaction = Bytes.ofUnsignedLong(transactionGasUsed);
+        receipts.add(receipt);
+        ObjectNode receiptObject = receiptsArray.addObject();
+        receiptObject.put(
+            "root", receipt.getStateRoot() == null ? "0x" : receipt.getStateRoot().toHexString());
+        receiptObject.put("status", "0x" + receipt.getStatus());
+        receiptObject.put("cumulativeGasUsed", Bytes.ofUnsignedLong(gasUsed).toQuantityHexString());
+        receiptObject.put("logsBloom", receipt.getBloomFilter().toHexString());
+        if (result.getLogs().isEmpty()) {
+          receiptObject.putNull("logs");
+        } else {
+          ArrayNode logsArray = receiptObject.putArray("logs");
+          for (Log log : result.getLogs()) {
+            logsArray.addPOJO(log);
+          }
+        }
+        receiptObject.put("transactionHash", transaction.getHash().toHexString());
+        receiptObject.put(
+            "contractAddress", transaction.contractAddress().orElse(Address.ZERO).toHexString());
+        receiptObject.put("gasUsed", gasUsedInTransaction.toQuantityHexString());
+        receiptObject.put("blockHash", Hash.ZERO.toHexString());
+        receiptObject.put("transactionIndex", Bytes.ofUnsignedLong(i).toQuantityHexString());
+      }
+    }
+
+    final ObjectNode resultObject = objectMapper.createObjectNode();
+
+    // block reward
+    if (!validTransactions.isEmpty()) {
+      worldStateUpdater
+          .getOrCreateSenderAccount(blockHeader.getCoinbase())
+          .getMutable()
+          .incrementBalance((reward == null) ? protocolSpec.getBlockReward() : reward);
+    }
+
+    // deposit CL withdrawals
```

Comment:
that sentence is going to be confusing when we add in-protocol validator deposits 😅

Reply:
rewritten.